### ISSUE TITLE
[Feature] Provide data resources.

### ISSUE THREAD

**Issue by stobias123** [F5Networks/terraform-provider-bigip#349]
There are many cases in which there are pre-existing resources in the F5 that I need to pull.

Please provide data resources for f5 objects.

Example.
```hcl
data "bigip_ssl_certificate" "cert" {
    name          = "test-clientssl"
    partition     = "Common"
}

# A Virtual server with SSL enabled
resource "bigip_ltm_virtual_server" "https" {
  name                       = "/Common/terraform_vs_https"
  destination                = var.vip_ip
  description                = "VirtualServer-test"
  port                       = 443
  pool                       = var.pool
  profiles                   = ["/Common/tcp",  "/Common/http",  data.bigip_ssl_certificate.cert.name ]
...
```

Outputs:
path to resource.

**Comment by nmenant** [comment]
Thanks for your feedback. 

Do you have specific data resources in mind you're interested in ? 

- Certs ? 
- iRules ? 
- DataGroup ? 
- monitors ? 
- others ?

**Comment by stobias123** [comment]
all of the above. read is implemented as part of nearly all resources, why not just take that method and implement as data resources?

**Comment by papineni87** [comment]
@stobias123 added some of the data sources in v1.5.0 release. ( certs, irules, datagroup, monitors )

**Comment by stobias123** [comment]
@papineni87  - This is great thank you. 

I think I'm going to take some of your work and add a few more. I've got some immediate applications for these data resources.

**Comment by stobias123** [comment]
@papineni87  - if resources are not found, the provider crashes..... These need error handling

**Comment by papineni87** [comment]
Issue Addressed in v1.6.0
